### DISCUSSION TITLE
How to assign new global variables in CWatM?

### DISCUSSION THREAD

**Discussion by jane-627** [iiasa/CWatM#24] [Q&A] [answered]
Hello, I wonder if I should do more to assign new global variables in CWatM. I've written these code in initial module in waterdemand.py(attached below):
self.var.irrPaddyRain = globals.inZero.copy()
self.var.irrNonpaddyRain = globals.inZero.copy()
and these code in dynamic module in waterdemand.py:
self.var.irrPaddyRain = self.var.fracVegCover[2] * self.var.Rain
self.var.irrNonpaddyRain = self.var.fracVegCover[3] * self.var.Rain

However, when I tried to run CWatM, Error 132 came out:
Error 132: Variable "irrPaddyRain" is not defined in "output_out_map_daily"
Closest variable to this name is: "irrPaddyDemand"

What should I do?
Thank you very much!
Jane
[waterdemand.zip](https://github.com/iiasa/CWatM/files/7115293/waterdemand.zip)

**Comment by cele5425** [comment]
Did you find an answer to this question?

**Comment by mikhailsmilovic** [comment, marked as answer]
Guten Tag, what you have is correct: the new variables are initialised and updated. This should work.

Trying this similarly, in the **develop branch**, I have created these two variables (similar to what you have done), and the outputs work.
- Rain_times_fracPaddy
- Rain_times_fracNonPaddy

This is in the commit "Example new variable" on Nov 14, 2023. 
Let me know what you think!

**Comment by jane-627** [comment]
Thank you very much！发自我的 iPhone在 2023年11月14日，23:21，Mikhail Smilovic ***@***.***> 写道：﻿
Guten Tag, what you have is correct: the new variables are initialised and updated. This should work.
Trying this similarly, in the develop branch, I have created these two variables (similar to what you have done), and the outputs work.

Rain_times_fracPaddy
Rain_times_fracNonPaddy

This is in the commit "Example new variable" on Nov 14, 2023.
Let me know what you think!
